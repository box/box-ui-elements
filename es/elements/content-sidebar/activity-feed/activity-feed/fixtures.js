export const replies = [{
  created_at: '2020-09-20T00:00:00Z',
  created_by: {
    id: '1',
    login: 'johndoe',
    name: 'John Doe',
    type: 'user'
  },
  id: '2485934',
  message: 'First reply',
  tagged_message: 'First reply',
  modified_at: '2020-09-20T00:00:00Z',
  modified_by: {
    id: '1',
    login: 'johndoe',
    name: 'John Doe',
    type: 'user'
  },
  permissions: {
    can_delete: true,
    can_edit: true,
    can_reply: false,
    can_resolve: true
  },
  type: 'comment'
}, {
  created_at: '2020-09-20T00:00:00Z',
  created_by: {
    id: '1',
    login: 'johndoe',
    name: 'John Doe',
    type: 'user'
  },
  id: '24835935',
  message: 'Last reply',
  tagged_message: 'Last reply',
  modified_at: '2020-09-20T00:00:00Z',
  modified_by: {
    id: '1',
    login: 'johndoe',
    name: 'John Doe',
    type: 'user'
  },
  permissions: {
    can_delete: true,
    can_edit: true,
    can_reply: false,
    can_resolve: true
  },
  type: 'comment'
}];
export default {
  replies
};
//# sourceMappingURL=fixtures.js.map