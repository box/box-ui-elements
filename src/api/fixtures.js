// @flow
import { TASK_NEW_NOT_STARTED } from '../constants';

export const task = {
    created_by: {
        type: 'task_collaborator',
        target: { name: 'Jay-Z', id: '100' },
        id: '000',
        role: 'CREATOR',
        status: TASK_NEW_NOT_STARTED,
    },
    created_at: '2019-01-01',
    due_at: '2019-02-02',
    id: '0',
    name: 'task message',
    type: 'task',
    assigned_to: {
        entries: [
            {
                id: '1',
                target: { name: 'Beyonce', id: '2', avatar_url: '', type: 'user' },
                status: TASK_NEW_NOT_STARTED,
                permissions: {
                    can_delete: false,
                    can_update: false,
                },
                role: 'ASSIGNEE',
                type: 'task_collaborator',
            },
        ],
        limit: 10,
        next_marker: null,
    },
    permissions: {
        can_update: false,
        can_delete: false,
        can_create_task_collaborator: false,
        can_create_task_link: false,
    },
    task_links: {
        entries: [
            {
                id: '03',
                type: 'task_link',
                target: {
                    type: 'file',
                    id: '4',
                },
                permissions: {
                    can_delete: false,
                    can_update: false,
                },
            },
        ],
        limit: 1,
        next_marker: null,
    },
    status: TASK_NEW_NOT_STARTED,
};

export const threadedComments = [
    {
        created_at: '1970-01-01T00:00:00.001Z',
        created_by: {
            id: '10',
            type: 'user',
            name: 'u1_name',
            login: 'u1@box.com',
        },
        id: '20',
        item: {
            type: 'file',
            id: 'f1',
        },
        message: '@[111:Aaron Levie] these tigers are cool!',
        modified_at: '1970-01-01T00:00:00.001Z',
        type: 'comment',
        parent: null,
        permissions: {
            can_delete: false,
            can_edit: false,
            can_reply: true,
            can_resolve: true,
        },
        replies: [
            {
                created_at: '1970-01-01T00:00:00.002Z',
                created_by: {
                    id: '11',
                    type: 'user',
                    name: 'u2_name',
                    login: 'u2@box.com',
                },
                id: '21',
                item: {
                    type: 'file',
                    id: 'f1',
                },
                message: '@[u1:Mateusz Mamczarz] Yes, they really are!',
                modified_at: '1970-01-01T00:00:00.002Z',
                type: 'comment',
                parent: {
                    id: 'c1',
                    type: 'comment',
                },
                permissions: {
                    can_delete: false,
                    can_edit: false,
                    can_reply: false,
                    can_resolve: true,
                },
                replies: [],
                total_reply_count: 0,
                status: 'open',
            },
        ],
        total_reply_count: 1,
        status: 'open',
    },
    {
        created_at: '1970-01-01T00:00:00.003Z',
        created_by: {
            id: '12',
            type: 'user',
            name: 'u3_name',
            login: 'u3@box.com',
        },
        id: '21',
        item: {
            type: 'file',
            id: 'f1',
        },
        message: 'Test message',
        modified_at: '1970-01-01T00:00:00.003Z',
        type: 'comment',
        parent: null,
        permissions: {
            can_delete: false,
            can_edit: false,
            can_reply: true,
            can_resolve: true,
        },
        replies: [],
        total_reply_count: 0,
        status: 'open',
    },
];

export default {
    task,
    threadedComments,
};
