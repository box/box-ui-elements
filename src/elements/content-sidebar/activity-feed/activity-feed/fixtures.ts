export const replies = [
    {
        created_at: '2020-09-20T00:00:00Z',
        created_by: {
            id: '1',
            login: 'johndoe',
            name: 'John Doe',
            type: 'user' as const,
        },
        id: '2485934',
        message: 'First reply',
        tagged_message: 'First reply',
        modified_at: '2020-09-20T00:00:00Z',
        modified_by: {
            id: '1',
            login: 'johndoe',
            name: 'John Doe',
            type: 'user' as const,
        },
        permissions: {
            can_delete: true,
            can_edit: true,
            can_reply: false,
            can_resolve: true,
        },
        type: 'comment' as const,
    },
    {
        created_at: '2020-09-20T00:00:00Z',
        created_by: {
            id: '1',
            login: 'johndoe',
            name: 'John Doe',
            type: 'user' as const,
        },
        id: '24835935',
        message: 'Last reply',
        tagged_message: 'Last reply',
        modified_at: '2020-09-20T00:00:00Z',
        modified_by: {
            id: '1',
            login: 'johndoe',
            name: 'John Doe',
            type: 'user' as const,
        },
        permissions: {
            can_delete: true,
            can_edit: true,
            can_reply: false,
            can_resolve: true,
        },
        type: 'comment' as const,
    },
];

export default { replies };
