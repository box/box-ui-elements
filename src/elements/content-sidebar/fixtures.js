// @flow
import {
    FEED_ITEM_TYPE_ANNOTATION,
    FEED_ITEM_TYPE_COMMENT,
    FEED_ITEM_TYPE_TASK,
    FEED_ITEM_TYPE_VERSION,
    TASK_NEW_NOT_STARTED,
} from '../../constants';

export const filterableActivityFeedItems = {
    annotationOpen: {
        type: FEED_ITEM_TYPE_ANNOTATION,
        id: 'open2',
        tagged_message: '',
        message: 'test',
        created_at: '2022-07-26T09:08:20-07:00',
        created_by: {
            type: 'user',
            id: '6187936317',
            name: 'Jhon',
            login: 'jdoe@box.com',
        },
        modified_at: '2022-07-26T09:08:20-07:00',
        permissions: {
            can_delete: true,
            can_edit: true,
            can_reply: true,
        },
        status: 'open',
    },
    annotationResolved: {
        type: FEED_ITEM_TYPE_ANNOTATION,
        id: 'open2',
        tagged_message: '',
        message: 'test',
        created_at: '2022-07-26T09:08:20-07:00',
        created_by: {
            type: 'user',
            id: '6187936317',
            name: 'Jhon',
            login: 'jdoe@box.com',
        },
        modified_at: '2022-07-26T09:08:20-07:00',
        permissions: {
            can_delete: true,
            can_edit: true,
            can_reply: true,
        },
        status: 'resolved',
    },
    commentOpen: {
        type: FEED_ITEM_TYPE_COMMENT,
        id: 'open1',
        tagged_message: '',
        message: 'test',
        created_at: '2022-07-26T09:08:20-07:00',
        created_by: {
            type: 'user',
            id: '6187936317',
            name: 'Jhon',
            login: 'jdoe@box.com',
        },
        modified_at: '2022-07-26T09:08:20-07:00',
        permissions: {
            can_delete: true,
            can_edit: true,
            can_reply: true,
        },
        status: 'open',
    },
    commentResolved: {
        type: FEED_ITEM_TYPE_COMMENT,
        id: 'open1',
        tagged_message: '',
        message: 'test',
        created_at: '2022-07-26T09:08:20-07:00',
        created_by: {
            type: 'user',
            id: '6187936317',
            name: 'Jhon',
            login: 'jdoe@box.com',
        },
        modified_at: '2022-07-26T09:08:20-07:00',
        permissions: {
            can_delete: true,
            can_edit: true,
            can_reply: true,
        },
        status: 'resolved',
    },
    taskItem: {
        created_by: {
            type: 'task_collaborator',
            target: { name: 'Jay-Z', id: '100' },
            id: '000',
            role: 'CREATOR',
            status: TASK_NEW_NOT_STARTED,
        },
        created_at: '2019-01-01',
        due_at: '2019-02-02',
        id: '0',
        name: 'task message',
        type: FEED_ITEM_TYPE_TASK,
        assigned_to: {
            entries: [
                {
                    id: '1',
                    target: { name: 'Beyonce', id: '2', avatar_url: '', type: 'user' },
                    status: TASK_NEW_NOT_STARTED,
                    permissions: {
                        can_delete: false,
                        can_update: false,
                    },
                    role: 'ASSIGNEE',
                    type: 'task_collaborator',
                },
            ],
            limit: 10,
            next_marker: null,
        },
        permissions: {
            can_update: false,
            can_delete: false,
            can_create_task_collaborator: false,
            can_create_task_link: false,
        },
        task_links: {
            entries: [
                {
                    id: '03',
                    type: 'task_link',
                    target: {
                        type: 'file',
                        id: '4',
                    },
                    permissions: {
                        can_delete: false,
                        can_update: false,
                    },
                },
            ],
            limit: 1,
            next_marker: null,
        },
        status: TASK_NEW_NOT_STARTED,
    },
    versionItem: {
        type: FEED_ITEM_TYPE_VERSION,
        id: '1060370614597',
        authenticated_download_url: 'https://someurl.com',
        created_at: '2022-07-06T03:01:28-07:00',
        extension: 'boxnote',
        is_download_available: true,
        modified_at: '2022-07-06T03:01:28-07:00',
        modified_by: {
            type: 'user',
            id: '18836063940',
            name: 'John Doe',
            login: 'jdoe@box.com',
        },
        name: 'Title',
        permissions: {
            can_download: true,
            can_preview: true,
            can_delete: true,
            can_annotate: false,
            can_view_annotations_all: true,
            can_view_annotations_self: true,
            can_create_annotations: false,
            can_view_annotations: false,
        },
        restored_at: null,
        restored_by: null,
        retention: null,
        size: 4159,
        trashed_at: null,
        trashed_by: null,
        uploader_display_name: 'John Doe',
        version_number: '27',
    },
};

export default {
    filterableActivityFeedItems,
};
