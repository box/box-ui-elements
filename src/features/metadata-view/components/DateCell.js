/** @flow */
import * as React from 'react';

import type { CellData } from '../../metadata-instance-editor/flowTypes';

import '../styles/MetadataViewCell.scss';

type Props = {
    cellData: CellData,
    intl: Object,
};

const DateCell = ({ cellData, intl }: Props) => {
    return (
        <div className="cell-container">
            <span>
                {intl.formatDate(cellData.cellData, {
                    month: 'long',
                    day: '2-digit',
                    year: 'numeric',
                })}
            </span>
        </div>
    );
};
export default DateCell;
