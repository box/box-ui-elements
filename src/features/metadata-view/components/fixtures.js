const visibleColumns = [
    {
        id: 'item_27',
        isChecked: true,
        label: 'Size',
        key: 'size',
        displayName: 'Size',
    },
    {
        id: 'item_28',
        isChecked: true,
        label: 'Vendor Name',
        key: 'vendor name',
        displayName: 'Vendor Name',
    },
    {
        id: 'item_26',
        isChecked: true,
        label: 'Updated',
        key: 'updated',
        displayName: 'Updated',
    },
    {
        id: 'item_29',
        isChecked: true,
        label: 'Contract Value',
        key: 'contract value',
        displayName: 'Contract Value',
    },
    {
        id: 'item_30',
        isChecked: true,
        label: 'Expiration Month',
        key: 'expiration month',
        displayName: 'Expiration Month',
    },
    {
        id: 'item_31',
        isChecked: true,
        label: 'Country',
        key: 'country',
        displayName: 'Country',
    },
    {
        id: 'item_32',
        isChecked: true,
        label: 'State',
        key: 'state',
        displayName: 'State',
    },
    {
        id: 'item_33',
        isChecked: true,
        label: 'Function',
        key: 'function',
        displayName: 'Function',
    },
];

const expectedVisibleColumns = {
    visibleColumns: [
        {
            displayName: 'Size',
            id: 'item_27',
            key: 'size',
            type: 'integer',
        },
        {
            displayName: 'Vendor Name',
            id: 'item_28',
            key: 'vendor name',
            type: 'string',
        },
        {
            displayName: 'Updated',
            id: 'item_26',
            key: 'updated',
            type: 'enum',
        },
        {
            displayName: 'Contract Value',
            id: 'item_29',
            key: 'contract value',
            type: 'enum',
        },
        {
            displayName: 'Expiration Month',
            id: 'item_30',
            key: 'expiration month',
            type: 'enum',
        },
        {
            displayName: 'Country',
            id: 'item_31',
            key: 'country',
            type: 'string',
        },
        {
            displayName: 'State',
            id: 'item_32',
            key: 'state',
            type: 'string',
        },
        {
            displayName: 'Function',
            id: 'item_33',
            key: 'function',
            type: 'string',
        },
    ],
};

const expectedVisibleColumnsOneHidden = {
    visibleColumns: [
        {
            displayName: 'Size',
            id: 'item_27',
            key: 'size',
            type: 'integer',
        },
        {
            displayName: 'Vendor Name',
            id: 'item_28',
            key: 'vendor name',
            type: 'string',
        },
        {
            displayName: 'Updated',
            id: 'item_26',
            key: 'updated',
            type: 'enum',
        },
        {
            displayName: 'Contract Value',
            id: 'item_29',
            key: 'contract value',
            type: 'enum',
        },
        {
            displayName: 'Expiration Month',
            id: 'item_30',
            key: 'expiration month',
            type: 'enum',
        },
        {
            displayName: 'Country',
            id: 'item_31',
            key: 'country',
            type: 'string',
        },
        {
            displayName: 'State',
            id: 'item_32',
            key: 'state',
            type: 'string',
        },
    ],
};

const visibleColumnsOneHidden = [
    {
        id: 'item_27',
        isChecked: true,
        label: 'Size',
        key: 'size',
        displayName: 'Size',
    },
    {
        id: 'item_28',
        isChecked: true,
        label: 'Vendor Name',
        key: 'vendor name',
        displayName: 'Vendor Name',
    },
    {
        id: 'item_26',
        isChecked: true,
        label: 'Updated',
        key: 'updated',
        displayName: 'Updated',
    },
    {
        id: 'item_29',
        isChecked: true,
        label: 'Contract Value',
        key: 'contract value',
        displayName: 'Contract Value',
    },
    {
        id: 'item_30',
        isChecked: true,
        label: 'Expiration Month',
        key: 'expiration month',
        displayName: 'Expiration Month',
    },
    {
        id: 'item_31',
        isChecked: true,
        label: 'Country',
        key: 'country',
        displayName: 'Country',
    },
    {
        id: 'item_32',
        isChecked: true,
        label: 'State',
        key: 'state',
        displayName: 'State',
    },
    {
        id: 'item_33',
        isChecked: false,
        label: 'Function',
        key: 'function',
        displayName: 'Function',
    },
];

const visibleColumnsAfterMount = {
    visibleColumns: [
        {
            displayName: 'Size',
            id: 'item_27',
            key: 'size',
            type: 'integer',
        },
        {
            displayName: 'Vendor Name',
            id: 'item_28',
            key: 'vendor name',
            type: 'string',
        },
        {
            displayName: 'Updated',
            id: 'item_26',
            key: 'updated',
            type: 'enum',
        },
        {
            displayName: 'Contract Value',
            id: 'item_29',
            key: 'contract value',
            type: 'enum',
        },
        {
            displayName: 'Expiration Month',
            id: 'item_30',
            key: 'expiration month',
            type: 'enum',
        },
        {
            displayName: 'Country',
            id: 'item_31',
            key: 'country',
            type: 'string',
        },
        {
            displayName: 'State',
            id: 'item_32',
            key: 'state',
            type: 'string',
        },
    ],
    widths: {
        size: 0.14285714285714285,
        'vendor name': 0.14285714285714285,
        updated: 0.14285714285714285,
        'contract value': 0.14285714285714285,
        'expiration month': 0.14285714285714285,
        country: 0.14285714285714285,
        state: 0.14285714285714285,
    },
};

const dateCellData = {
    cellData: '2016-02-22T00:00:00.000Z',
    columnData: undefined,
    columnIndex: 3,
    dataKey: 'name',
    isScrolling: false,
    parent: {},
    rowData: { name: 'Gnarly Dude!', stuff: 'and things!' },
};

const enumCellData = {
    cellData: 'Yaharg Matey',
    columnData: undefined,
    columnIndex: 3,
    dataKey: 'name',
    isScrolling: false,
    parent: {},
    rowData: { name: 'Gnarly Dude!', stuff: 'and things!' },
};

const floatCellData = {
    cellData: 'yoyo',
    columnData: undefined,
    columnIndex: 3,
    dataKey: 'name',
    isScrolling: false,
    parent: {},
    rowData: { name: 'Gnarly Dude!', stuff: 'and things!' },
};

const integerCellData = {
    cellData: 3,
    columnData: undefined,
    columnIndex: 3,
    dataKey: 'name',
    isScrolling: false,
    parent: {},
    rowData: { name: 'Gnarly Dude!', stuff: 'and things!' },
};

const multiSelectCellData = {
    cellData: 'Yaharg Matey',
    columnData: undefined,
    columnIndex: 3,
    dataKey: 'name',
    isScrolling: false,
    parent: {},
    rowData: { name: 'Gnarly Dude!', stuff: 'and things!' },
};

const nameCellData = {
    cellData: 3,
    columnData: undefined,
    columnIndex: 3,
    dataKey: 'name',
    isScrolling: false,
    parent: {},
    rowData: { name: 'Gnarly Dude!', stuff: 'and things!' },
};

const stringCellData = {
    cellData: 'yoyo',
    columnData: undefined,
    columnIndex: 3,
    dataKey: 'name',
    isScrolling: false,
    parent: {},
    rowData: { name: 'Gnarly Dude!', stuff: 'and things!' },
};

const template = {
    id: 'template1',
    templateKey: 'template1',
    displayName: 'template1 title',
    scope: 'enterprise_123',
    'Vendor Name': {
        operators: ['is', 'is greater than', 'is less than', 'is not', 'is blank', 'matches any'],
        values: ['Google', 'Apple', 'Facebook'],
    },
    'Expiration Month': {
        operators: ['is', 'is greater than', 'is less than', 'is not'],
        values: ['August 2018', 'September 2018', 'October 2018'],
    },
    'File Type': {
        operators: ['is', 'is not'],
        values: ['.docx', '.mp3', 'mp4'],
    },
    fields: [
        {
            id: 'item_27',
            type: 'integer',
            key: 'size',
            displayName: 'Size',
        },
        {
            id: 'item_28',
            type: 'string',
            key: 'vendor name',
            displayName: 'Vendor Name',
        },
        {
            id: 'item_26',
            type: 'enum',
            key: 'updated',
            displayName: 'Updated',
        },
        {
            id: 'item_29',
            type: 'enum',
            key: 'contract value',
            displayName: 'Contract Value',
        },
        {
            id: 'item_30',
            type: 'enum',
            key: 'expiration month',
            displayName: 'Expiration Month',
        },
        {
            id: 'item_31',
            type: 'string',
            key: 'country',
            displayName: 'Country',
        },
        {
            id: 'item_32',
            type: 'string',
            key: 'state',
            displayName: 'State',
        },
        {
            id: 'item_33',
            type: 'string',
            key: 'function',
            displayName: 'Function',
        },
    ],
    filesList: [{ name: '1' }, { name: '2' }, { name: '3' }],
    templateName: 'Vendor Contracts',
};

const instances = [
    {
        canEdit: true,
        id: 'editor1',
        data: {
            contractValue: 3,
            fileType: 'pdf',
            name: 'google',
            lastModified: '2018-06-20T00:00:00.000Z',
            size: 1,
        },
        cascadePolicy: {
            canEdit: true,
            isEnabled: false,
            id: 'some cascading policy id',
        },
    },
    {
        canEdit: true,
        id: 'editor2',
        data: {
            contractValue: 34,
            fileType: 'powerpoint-presentation',
            name: 'facebook',
            lastModified: '2018-06-20T00:00:00.000Z',
            size: 12,
        },
        cascadePolicy: {
            canEdit: true,
            isEnabled: false,
            id: 'some cascading policy id',
        },
    },
    {
        canEdit: true,
        id: 'editor3',
        data: {
            contractValue: 5,
            fileType: 'video',
            name: 'amazon',
            lastModified: '2018-06-20T00:00:00.000Z',
            size: 45,
        },
        cascadePolicy: {
            canEdit: true,
            isEnabled: false,
            id: 'some cascading policy id',
        },
    },
];

const headerData = {
    columnData: {},
    dataKey: 'blargh',
    disableSort: false,
    label: 'Yahargh',
    sortBy: {},
    SortDirection: {},
};

const metadataViewProps = {
    totalWidth: 700,
    tableHeight: 300,
    tableHeaderHeight: 40,
    tableRowHeight: 50,
    widths: {
        icon: 0.1,
        name: 0.4,
        lastModified: 0.6,
        size: 0.3,
        contractValue: 0.3,
    },
};

const metadataTableStateAfterMount = {
    visibleColumns: [
        {
            displayName: 'Size',
            id: 'item_27',
            key: 'size',
            type: 'integer',
        },
        {
            displayName: 'Vendor Name',
            id: 'item_28',
            key: 'vendor name',
            type: 'string',
        },
        {
            displayName: 'Updated',
            id: 'item_26',
            key: 'updated',
            type: 'enum',
        },
        {
            displayName: 'Contract Value',
            id: 'item_29',
            key: 'contract value',
            type: 'enum',
        },
        {
            displayName: 'Expiration Month',
            id: 'item_30',
            key: 'expiration month',
            type: 'enum',
        },
        {
            displayName: 'Country',
            id: 'item_31',
            key: 'country',
            type: 'string',
        },
        {
            displayName: 'State',
            id: 'item_32',
            key: 'state',
            type: 'string',
        },
        {
            displayName: 'Function',
            id: 'item_33',
            key: 'function',
            type: 'string',
        },
    ],
    widths: {
        size: 0.125,
        'vendor name': 0.125,
        updated: 0.125,
        'contract value': 0.125,
        'expiration month': 0.125,
        country: 0.125,
        state: 0.125,
        function: 0.125,
    },
};

const expectedColumnsChangedWidths = {
    'contract value': 0.14285714285714285,
    country: 0.14285714285714285,
    'expiration month': 0.14285714285714285,
    size: 0.14285714285714285,
    state: 0.14285714285714285,
    updated: 0.14285714285714285,
    'vendor name': 0.14285714285714285,
};

const expectedResizeRowWidths = {
    'contract value': 0.12452205410942857,
    country: 0.125,
    'expiration month': 0.125,
    function: 0.125,
    size: 0.125,
    state: 0.125,
    updated: 0.12547794589057143,
    'vendor name': 0.125,
};

export {
    visibleColumns,
    expectedVisibleColumns,
    visibleColumnsOneHidden,
    expectedVisibleColumnsOneHidden,
    visibleColumnsAfterMount,
    dateCellData,
    enumCellData,
    floatCellData,
    integerCellData,
    nameCellData,
    multiSelectCellData,
    stringCellData,
    template,
    instances,
    headerData,
    metadataViewProps,
    metadataTableStateAfterMount,
    expectedColumnsChangedWidths,
    expectedResizeRowWidths,
};
