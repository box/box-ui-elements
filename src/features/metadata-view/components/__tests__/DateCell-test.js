import * as React from 'react';

import { dateCellData } from '../fixtures';

import DateCell from '../DateCell';

const intlShape = {
    formatDate: () => 'February 2016',
};

describe('features/metadata-view/components/DateCell', () => {
    test('should render', () => {
        const wrapper = shallow(<DateCell cellData={dateCellData} intl={intlShape} />);
        expect(wrapper).toMatchSnapshot();
    });
});
