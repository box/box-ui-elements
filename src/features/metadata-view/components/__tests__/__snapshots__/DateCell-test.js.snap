// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/metadata-view/components/DateCell should render 1`] = `
<div
  className="cell-container"
>
  <span>
    February 2016
  </span>
</div>
`;
